crates/fuel-core/src/schema/balance.rs

```diff
@@ -65,10 +67,9 @@ impl BalanceQuery {
         Ok(balance)
     }
 
-    // TODO: We can't paginate over `AssetId` because it is not unique.
-    //  It should be replaced with `UtxoId`.
-    //  This API should be migrated to the indexer for better support and
+    // TODO: This API should be migrated to the indexer for better support and
     //  discontinued within fuel-core.
+    #[graphql(complexity = "QUERY_COSTS.balance_query")]
```

Comment:
Shouldn't this be multiplied by the number of assets? Ie if you request your first 10 balances, it should be 10 * balance_query

Reply:
The code inside iterates over all assets in the storage regardless of input parameters.